libsoftwarecontainer/src/gateway/devicenode/devicenodelogic.cpp

```diff
@@ -18,33 +18,40 @@
  */
 
 #include "devicenodelogic.h"
+#include "devicenodegateway.h"
 
 namespace softwarecontainer {
 
-std::vector<DeviceNodeParser::Device>::iterator
+std::vector<std::shared_ptr<Device>>::iterator
 DeviceNodeLogic::findDeviceByName(const std::string name)
 {
-    std::vector<DeviceNodeParser::Device>::iterator item =
+    std::vector<std::shared_ptr<Device>>::iterator item =
             std::find_if(m_devList.begin(), m_devList.end(),
-            [&] (DeviceNodeParser::Device const &d) { return d.name == name; });
+            [&] (std::shared_ptr<Device> const &d) { return d->name == name; });
 
     return item;
```

Comment:
I think it's a bit weird to return an iterator to a shared_ptr. Can't you return the shared_ptr directly instead?